### PR TITLE
fix(project): 中文标题不再塌成 slug 作为项目显示名

### PR DESCRIPTION
**Pull request by Pollo3470**
## Summary
- slug 算法在无 ASCII 字母前缀时统一塌为中性 `proj-<hex>`,避免「第1集」→ `1-55898f35`、「我的小说」→ `project-352ec63c` 等误导性 slug
- `create_project_metadata` 在 title 空时保留空字符串,不再把 slug 写进 title 字段;前端用 i18n `dashboard:untitled_project` 兜底显示「未命名项目」
- 放宽 `data_validator` title 校验(string 即可,允许空);`list_projects` 路由三处 fallback 同步改回空串
- 存量已被污染的 `title=slug` 项目不做自动迁移,用户在项目设置页手改

## Test plan
- [x] `uv run python -m pytest tests/test_project_manager_more.py tests/test_data_validator.py tests/test_projects_router.py tests/test_i18n_consistency.py`
- [x] `uv run ruff check . && uv run ruff format --check .`
- [x] `uv run basedpyright`(本次改动 0 errors)
- [x] `cd frontend && pnpm check`(typecheck + lint + 579 vitest 全绿)
- [x] 端到端:`generate_project_name("第1集")` → `proj-xxxxxxxx`、`create_project_metadata(name, "第1集")["title"] == "第1集"`、空 title 持久化为 `""`
- [ ] 手动 UI:创建中文标题项目,确认 URL `proj-xxx`、header 显示中文 title;切 EN/VI 看「未命名项目」三语

## 发布说明

* **Bug Fixes**
  * 统一前端项目标题展示：当标题为空或仅空白时一致显示“未命名项目”、更新相关提示与 aria 文案。
  * 后端列表接口与校验调整：项目元数据中缺失或非法标题不再回退为名称，空标题以空串返回/允许。

* **多语言支持**
  * 新增英文、越南文、中文的“未命名项目”翻译条目。

* **Tests**
  * 增加与更新测试，覆盖空标题允许、显示回退和元数据生成行为。

[Review Change Stack](https://app.coderabbit.ai/change-stack/ArcReel/ArcReel/pull/641?utm_source=github_walkthrough&utm_medium=github&utm_campaign=change_stack)

### CODE REVIEW

File: lib/data_validator.py [hunk 1]
```diff
@@ -163,7 +163,7 @@ def _validate_project_payload(
         errors: list[str],
         warnings: list[str],
     ) -> None:
-        if not project.get("title"):
+        if "title" not in project or not isinstance(project["title"], str):
             errors.append("缺少必填字段: title")
```
[the remark this answers is not in the document]
Comment (Pollo3470) [reply lib/data_validator.py]: 已拆分两条 `if`:title 缺失保留原文案 `缺少必填字段: title`,类型不为 str 报 `字段类型错误: title 应为字符串`。`tests/test_data_validator.py::test_validate_project_rejects_non_string_title` 覆盖回归。

File: server/routers/projects.py [hunk 2]
```diff
@@ -397,7 +397,9 @@ def _sync():
                     projects.append(
                         {
                             "name": name,
-                            "title": project.get("title", name),
+                            # title 缺失/空时返回空串,由前端 i18n 兜底显示「未命名项目」,
+                            # 避免把 slug 风格的 name 当成显示名暴露给用户。
+                            "title": project.get("title", ""),
```
[the remark this answers is not in the document]
Comment (Pollo3470) [reply server/routers/projects.py]: 三处 fallback 改用 `isinstance(raw_title, str) else ""`:title 字段为 `None` 或非 str 时统一归一为空串,确保接口契约始终返回 `str`。没沿用 `dict.get` 的 default 兜底,因为它对"键存在但值为 None"无效。

File: lib/project_manager.py [hunk 3]
```diff
@@ -131,10 +131,17 @@ def normalize_project_name(name: str) -> str:
...
+        CJK 标题经 NFKD + ascii ignore 后会丢光汉字;若结果不包含任何字母
+        （纯空 / 仅夹在标题里的孤立数字,如「第1集」→ "1"),退化为中性
+        前缀 ``proj``,避免产生 ``1-<hex>`` 这种看似有意义实则误导的 slug。
+        """
         ascii_text = unicodedata.normalize("NFKD", str(title).strip()).encode("ascii", "ignore").decode("ascii")
         slug = PROJECT_SLUG_SANITIZER.sub("-", ascii_text).strip("-_").lower()
-        return slug[:24] or "project"
+        if not slug or not any(c.isalpha() for c in slug):
+            return "proj"
+        return slug[:24]
```
[the remark this answers is not in the document]
Comment (Pollo3470) [reply lib/project_manager.py]: 已调换为 sanitize → truncate → validate:`slug = ...[:24]` 提到 letter 校验之前。corner case 例如 `"1-234567890-1234567890-1234-letters"` 截前 24 字符 `"1-234567890-1234567890-1"` 只剩数字/连字符 → 塌成 `proj`。回归测试 `tests/test_project_manager_more.py::test_generate_project_name_truncates_before_letter_check` 覆盖。

File: server/routers/projects.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (Pollo3470) [reply server/routers/projects.py]: 方向采纳,具体写法换成 `raw_title if isinstance(raw_title, str) else ""`,没用 `str(project.get("title") or "")`。理由:`str(None)` 会被 `or ""` 兜住,但 `str(123)` 会产生 `"123"` 这种 stringify,破坏"非 str 一律归零"的契约,在数据上游异常时反而把脏值抛到前端。`isinstance` 判定更严格也更显式。
